app/controllers/gift_requests_controller.rb

```diff
@@ -48,6 +48,19 @@ def catalog
     @gift_requests = @event.gift_requests.unreserved.order(created_at: :asc).page(params[:page] || 1)
   end
 
+
+  def reserve
+    @gift_request = GiftRequest.find(params[:id])
+    reservation = Reservation.new(gift_request_id: params[:id], donor_id: current_user.id)
+    if reservation.save
+      redirect_to wishlist_path(@gift_request.list.id)
+    else
+      # render "gift_request"
+      render status: 500
```

Comment:
I think at the moment, we just redirect the user back to the same page, at least for `save` calls on the therapist's side. We could probably do the same and just populate an error message of some sort? I'm not 100% sure what would be best here.